### DISCUSSION TITLE
Access OpenAPI description in handler

### DISCUSSION THREAD

**Discussion by bachp** [juhaku/utoipa#1064] [Q&A] [answered]
Is it possible to access the associated OpenAPI definition from within the associated handler?

For example, in the following example, I would like to access the `responses` data from within `get_hello`.

```
#[derive(OpenApi)]
#[openapi(paths(get_hello))]
struct ApiDoc;

#[utoipa::path(
    get,
    path = "/hello",
    responses(
        (status = 200, description = "Hello response")
    )
)]
async fn get_hello() -> &'static str {
    "Hello, World!"
}
```

Is this possible?

**Comment by juhaku** [comment, marked as answer]
There are 2 ways around this. 
1. Pass around the `OpenApi` type by some means. E.g. via global state or whatever state mechanism your framework will support. 
2. The `#[utoipa::path(...)]` macro will create a meta struct called `__path_get_hello` for the annotated function. In general in form of `__path_{function_name}`. Thus you could directly call this struct with `__path_get_hello::operation()` to get the generated operation schema to access it's content. However there is no way to change this content here. 

Only way to change the content is to access via `OpenApi` directly or registered `Modify` trait.

**Reply by bachp** [reply]
To be more precise you can access the `PathItem` struct using:

```rust
use utoipa::Path;
let path_item = __path_post_login::path_item(None);

// now access it's fields
path_item.summary;
```

**Reply by juhaku** [reply]
Ah, yes, good catch, that is the currently supported syntax. I already forgot that the syntax will change with the coming 5.0.0 release.
